CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -299,6 +299,17 @@ function PlayerHospital:onEndDay()
     self:dailyAdviceChecks()
   end
 
+  -- Occasionally unstaffed rooms with a queue will call for staff
+  for _, room in pairs(self.world.rooms) do
+    if math.random(1, 20) == 1 and #room.door.queue > 0 and not room.staff_member and
+        not room:testStaffCriteria(room:getRequiredStaffCriteria()) then
+          self.world.dispatcher:callForStaff(room)
+          break
+    end
+  end
```

Comment:
The whole point of the dispatcher is that you don't need this. That is, if all is well, the dispatcher should already know the room needs staff.

That said, I am not convinced that the dispatcher is the right solution. I'd prefer something much simpler, where things that need other things just look for the latter directly.
It avoids the whole administration of needs, and cleanup of callbacks which is a source of bugs.

If you want to go this direction, it would be a separate patch.